llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -297,6 +298,19 @@ void collectFunctionsAndGlobalVariablesToExtract(
   }
 }
 
+// Checks for use of undefined user functions and emits a warning message.
+void checkForCallsToUndefinedFunctions(const Module &M) {
+  if (AllowDeviceImageDependencies)
+    return;
+  for (const Function &F : M) {
+    if (!F.isIntrinsic() && !F.getName().starts_with("__") &&
```

Comment:
These checks ensure that we are looking at only user functions  and not LLVM builtins or SPIR-V builtins.

Thanks